app/controllers/api/v1/meetings_controller.rb

```diff
@@ -39,12 +50,25 @@ def status
         data = {
           status: BigBlueButtonApi.new.meeting_running?(room: @room)
         }
-        data[:joinUrl] = BigBlueButtonApi.new.join_meeting(room: @room, name: params[:name], role: 'Viewer') if data[:status]
 
-        if authorize_access?
-          render_json data:, status: :ok
-        else
-          render_json status: :unauthorized
+        case params[:access_code].blank?
+        when true
+          if @room.viewer_access_code.blank?
+            data[:joinUrl] = BigBlueButtonApi.new.join_meeting(room: @room, name: params[:name], role: 'Viewer') if data[:status]
+            render_json data:, status: :ok
+          else
+            render_json status: :unauthorized
+          end
+        when false
+          if @room.viewer_access_code == params[:access_code]
+            data[:joinUrl] = BigBlueButtonApi.new.join_meeting(room: @room, name: params[:name], role: 'Viewer') if data[:status]
+            render_json data:, status: :ok
+          elsif @room.moderator_access_code == params[:access_code]
+            data[:joinUrl] = BigBlueButtonApi.new.join_meeting(room: @room, name: params[:name], role: 'Moderator') if data[:status]
+            render_json data:, status: :ok
+          else
+            render_json status: :unauthorized
+          end
```

Comment:
Working on this.

Reply:
Yeah spend some more time on this. This is super complex and hard to read

Reply:
Try a completely different approach to the logic here

Reply:
To make your life easier, you can just render `render_json status: :unauthorized` as the final case if nothing matches